pkg/nodeipam/ipam/range_allocator.go

```diff
@@ -327,21 +324,55 @@ func filterOutServiceRange(clusterCIDRs []*net.IPNet, cidrSets []*cidrset.CidrSe
 	}
 }
 
+func (r *rangeAllocator) allocatePodCIDRs() ([]*net.IPNet, error) {
+	allocatedCIDRs := make([]*net.IPNet, len(r.cidrSets))
+	for idx := range r.cidrSets {
+		podCIDR, err := r.cidrSets[idx].AllocateNext()
+		if err != nil {
```

Comment:
The AllocateNext would be a dummy operation if failed so no need to release cidrs here? https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/pkg/nodeipam/ipam/cidrset/cidr_set.go#L214

Reply:
But we need to release ips from cidrSet[0] to cidrSet[idx - 1]?